ingestion/src/metadata/ingestion/source/pipeline/openlineage/metadata.py

```diff
@@ -0,0 +1,534 @@
+# metadata.py
+
+"""
+OpenLineage source to extract metadata from Kafka events
+"""
+import json
+import traceback
+from collections import defaultdict
+from itertools import groupby, product
+from typing import Any, Dict, Iterable, List, Optional, Tuple, Union
+
+from metadata.generated.schema.api.data.createPipeline import CreatePipelineRequest
+from metadata.generated.schema.api.data.createTable import CreateTableRequest
+from metadata.generated.schema.api.lineage.addLineage import AddLineageRequest
+from metadata.generated.schema.entity.data.databaseSchema import DatabaseSchema
+from metadata.generated.schema.entity.data.pipeline import Pipeline
+from metadata.generated.schema.entity.data.table import Column, Table
+from metadata.generated.schema.entity.services.connections.pipeline.openLineageConnection import (
+    OpenLineageConnection,
+)
+from metadata.generated.schema.entity.services.ingestionPipelines.status import (
+    StackTraceError,
+)
+from metadata.generated.schema.metadataIngestion.workflow import (
+    Source as WorkflowSource,
+)
+from metadata.generated.schema.type.entityLineage import (
+    ColumnLineage,
+    EntitiesEdge,
+    LineageDetails,
+    Source,
+)
+from metadata.generated.schema.type.entityReference import EntityReference
+from metadata.ingestion.api.models import Either
+from metadata.ingestion.api.steps import InvalidSourceException
+from metadata.ingestion.models.pipeline_status import OMetaPipelineStatus
+from metadata.ingestion.ometa.ometa_api import OpenMetadata
+from metadata.ingestion.source.pipeline.openlineage.models import (
+    EventType,
+    LineageEdge,
+    LineageNode,
+    OpenLineageEvent,
+    TableDetails,
+    TableFQN,
+)
+from metadata.ingestion.source.pipeline.openlineage.utils import (
+    FQNNotFoundException,
+    message_to_open_lineage_event,
+)
+from metadata.ingestion.source.pipeline.pipeline_service import PipelineServiceSource
+from metadata.utils import fqn
+from metadata.utils.logger import ingestion_logger
+
+logger = ingestion_logger()
+
+
+class OpenlineageSource(PipelineServiceSource):
+    """
+    Implements the necessary methods of PipelineServiceSource to facilitate registering OpenLineage pipelines with
+    metadata into Open Metadata.
+
+    Works under the assumption that OpenLineage integrations produce events to Kafka topic, which is a source of events
+    for this connector.
+
```

Comment:
@mgorsk1 do you mean that this would only support if the OpenLineage is configured with kafka as transport mechanism?

or does other transport mechanism like http would also be supported?

Reply:
that's right, these are two mutually exclusive transport types and this connector for now supports only kafka (as we are using it), it shouldn't be a problem for community to extend with api support as well - I suspect just configuration of connector would have to be extended + get pipelines method changed a bit.